controller/user/manager.py

```diff
@@ -104,53 +105,50 @@ def update_last_interaction(user_id: str) -> None:
-            active_user_by_id = active_users[user_id]
-            get_user["last_interaction"] = active_user_by_id["last_interaction"]
-            get_user["role"] = active_user_by_id["role"]
-            get_user["organization"] = active_user_by_id["organizationName"]
-
-            public_meta = get_user["metadata_public"]
-            get_user["sso_provider"] = (
-                public_meta.get("registration_scope", {}).get("provider_id", None)
-                if public_meta
-                else None
-            )
-
-            final_users.append(get_user)
-            save_len_final_users += 1
-
-    final_users = sorted(
-        final_users,
-        key=lambda x: (x[sort_key] is None, x.get(sort_key, "")),
-        reverse=sort_direction == -1,
-    )
-
-    # paginating users
-    final_users = final_users[offset : offset + limit]
-
-    return final_users, save_len_final_users
-
-
 def delete_user(user_id: str) -> None:
     user.delete(user_id, with_commit=True)
     user_activity.delete_user_activity(user_id, with_commit=True)
     kratos.__refresh_identity_cache()
+
+
+def migrate_kratos_users() -> None:
+    # this is only supposed to be called during startup of the application
+    daemon.run_with_db_token(__migrate_kratos_users)
+
+
+def __migrate_kratos_users():
+    users = kratos.get_cached_values()
+    filtered_users = {k: v for k, v in users.items() if k != "collected"}
+    users_database = user.get_by_id_list(list(filtered_users.keys()))
+    for user_database in users_database:
+        user_id = str(user_database.id)
+        user_identity = filtered_users[user_id]["identity"]
+        if (
+            user_database.email != user_identity["traits"]["email"]
+            or user_database.verified
+            != user_identity["verifiable_addresses"][0]["verified"]
+            or user_database.created_at
+            != user_identity["verifiable_addresses"][0]["created_at"]
+            or user_database.metadata_public != user_identity["metadata_public"]
+            or (
+                user_identity["metadata_public"]
+                .get("registration_scope", {})
+                .get("provider_id", None)
+                != user_database.sso_provider
+            )
+        ):
+            user.update_user(
+                user=user_database,
+                email=user_identity["traits"]["email"],
+                verified=user_identity["verifiable_addresses"][0]["verified"],
+                created_at=user_identity["verifiable_addresses"][0]["created_at"],
+                metadata_public=user_identity["metadata_public"],
+                sso_provider=(
+                    user_identity["metadata_public"]
+                    .get("registration_scope", {})
+                    .get("provider_id", None)
+                    if user_identity["metadata_public"]
+                    else None
+                ),
+                with_commit=True,
```

Comment:
commit on every change is quite expensive usually we do this after a loop (see comment above)